lib/core/utils/preload-media.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Given a rootNode
+ * -> get all HTMLMediaElement's and ensure their metadata is loaded
+ *
+ * @method preloadMedia
+ * @memberof axe.utils
+ * @property {Object} options.treeRoot (optional) the DOM tree to be inspected
+ */
+axe.utils.preloadMedia = function preloadMedia({ treeRoot = axe._tree[0] }) {
+	const mediaVirtualNodes = axe.utils.querySelectorAll(
+		treeRoot,
+		'video, audio'
+	);
+	if (!mediaVirtualNodes || !mediaVirtualNodes.length) {
```

Comment:
this is statement can be removed. `mediaVirtualNodes` is will always be an array, and an empty array will work fine with your map function. Promise.all will resolve.